workshops/interactive_maps/README.md

```diff
@@ -0,0 +1,279 @@
+```js
+function sayHello(){
+    console.log('Hello World!')
+}
+```
+These can now be called from our HTML file as addNumbers(someNumber, someNumber) and sayHello(). Now that you have a basic understanding of how functions work, head over into your script.js file and add this in.
+```js
+var map = document.getElementById('map');
+function changeImage(location){
+    if (location == 'ocean'){
+        map.src = 'https://cloud-9p8jb8vst.vercel.app/orgocean.png';
+    }
+    else if (location == 'residential'){
+        map.src = 'https://cloud-gga696gyn.vercel.app/orgresidential.png';
+    }
+    
+    else if (location == 'roadway'){
+        map.src = 'https://cloud-bzeicw4mm.vercel.app/orgroad.png';
+    }
+    
+    else if (location == 'vendors'){
+        map.src = "https://cloud-9ldvvvhb4.vercel.app/orgvendors.png";
+    }   
+    
+    else if (location == 'beach'){
+        map.src = 'https://cloud-3bvmpy2n1.vercel.app/orgbeach.png';
+    }
+    else if (location == 'normal'){
+        map.src = 'https://cloud-coh6o3mhy.vercel.app/orgbg.png';
+    }
+}
+```
+This snippet declares a variable named map, which corresponds to the map element in our html file. Then it creates a function named changeImage() with the parameter 'location'. When we call this function from our html file with specific values like 'ocean' or 'residential', (These values passed into parameters are called arguments.) they call on the actual function from the JS file. Inside the function, is an IF -ELSE IF statement. It does exactly what it means. If the location argument is 'beach', set the map's src attribute to the beach highlighted image, and so on. The syntax is
+```js
+if (condition) {
+    Do something;
+}
+else if (another condition) {
+    Do something else;
+}
+else {
+    Do something totally different;
+}
+```
+There are three basic If statements in Javascript. The normal If statement, the If - Else statement, and the If - Else If - Else statement. Not sure what that means? Let's go through some examples.
+```js
+var name = 'Shane';
+if (name == 'Shane'){
+  console.log('Hello Shane')
+}
+```
+This snippet shows a basic If statement. It first declares a variable called name which has the value of shane. Then the If statement proceeds to check out if the name is Shane, and if yes, it logs out 'Hello Shane'. Note that nothing happens if the name is not set to Shane. To try this for yourself, create a seperate repl with the language set to Javascript only. Paste that snippet and tweak the values. Also, it's time to talk about the equal to symbols. Why only one equal to at the first and two equal to-s at the bottom? This is because one equal to set's the value of the left hand side to the right hand side and two equal to-s compare both the values. Here, we have used an equal to and set the value of name to Shane. Then we have used two equal to-s to check **if** the name and Shane are both the same.
```

Comment:
```suggestion
This snippet shows a basic if statement. It first declares a variable called name which has the value of shane. Then the if statement proceeds to check out if the name is Shane, and if yes, it logs out 'Hello Shane'. Note that nothing happens if the name is not set to Shane. To try this for yourself, create a seperate repl with the language set to Javascript only. Paste that snippet and tweak the values. Also, it's time to talk about the equal to symbols. Why only one equal sign at the first and two equal signs at the bottom? This is because one equal sign set's the value of the left hand side to the right hand side and two equal signs compare both the values. Here, we have used an equal sign and set the value of name to Shane. Then we have used two equal signs to check **if** the name and Shane are both the same.
```